pkg/southbound/gnmi/conn.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"context"
+	"time"
+
+	"github.com/onosproject/onos-lib-go/pkg/certs"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-lib-go/pkg/uri"
+
+	"crypto/tls"
+
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	baseClient "github.com/openconfig/gnmi/client"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+var log = logging.GetLogger("southbound", "gnmi")
+
+// ConnID connection ID
+type ConnID string
+
+// Conn gNMI connection interface
+type Conn interface {
+	Client
+	Context() context.Context
```

Comment:
I'm not sure it makes sense to have a connection level `context.Context`. Contexts are meant to be limited to the scope of a request. We also need to be able to determine when the connection is closed by the _server_. We might need some other interface/method to detect disconnects to update the connection manager and reconcile the connection. We should look into the gRPC API and figure out how we can learn of connection state changes and then define the API from there.